model/src/main/java/uk/gov/pay/commons/api/validation/ExternalJsonMetadataValidator.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.pay.commons.api.validation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Iterator;
+import java.util.Map;
+
+public class ExternalJsonMetadataValidator implements ConstraintValidator<ValidExternalJsonMetadata, JsonNode> {
+
+    private static final int MAXIMUM_KEYS = 10;
+    private static final int MAXIMUM_VALUE_LENGTH = 50;
+    private static final int MAXIMUM_KEY_LENGTH = 30;
```

Comment:
is it possible to put all these private fields as part of a constructor? these values come from a requirement in pay-connector but this class is in pay-java-commons.

Reply:
I was thinking that what consider valid external metadata should be consistent across pay and all services. If we think its ok to allow the constraints to change out of sync across projects then that's a great idea and its easily done (we could permit those value to be set on the annotation and have the validator class read them in an overridden `init` method).

Reply:
ok